services/bundle_analysis.py

```diff
@@ -253,6 +265,129 @@ def process_upload(self, commit: Commit, upload: Upload) -> ProcessingResult:
             session_id=session_id,
         )
 
+    def _save_to_timeseries(
+        self,
+        db_session: Session,
+        commit: Commit,
+        name: str,
+        measurable_id: str,
+        value: float,
+    ):
+        command = insert(Measurement.__table__).values(
+            name=name,
+            owner_id=commit.repository.ownerid,
+            repo_id=commit.repoid,
+            measurable_id=measurable_id,
+            branch=commit.branch,
+            commit_sha=commit.commitid,
+            timestamp=commit.timestamp,
+            value=value,
+        )
+        command = command.on_conflict_do_update(
+            index_elements=[
+                Measurement.name,
+                Measurement.owner_id,
+                Measurement.repo_id,
+                Measurement.measurable_id,
+                Measurement.commit_sha,
+                Measurement.timestamp,
+            ],
+            set_=dict(
+                branch=command.excluded.branch,
+                value=command.excluded.value,
+            ),
+        )
+        db_session.execute(command)
+        db_session.flush()
+
+    @sentry_sdk.trace
+    def save_measurements(self, commit: Commit, upload: Upload) -> ProcessingResult:
+        """
+        Save timeseries measurements for this bundle analysis report
+        """
+        try:
+            commit_report: CommitReport = upload.report
+            repo_hash = ArchiveService.get_archive_hash(commit_report.commit.repository)
+            storage_service = get_storage_client()
+            bundle_loader = BundleAnalysisReportLoader(storage_service, repo_hash)
+
+            # fetch existing bundle report from storage
+            bundle_analysis_report = bundle_loader.load(commit_report.external_id)
+
+            dataset_names = [
+                dataset.name for dataset in repository_datasets_query(commit.repository)
+            ]
+
+            db_session = commit.get_db_session()
+            for bundle_report in bundle_analysis_report.bundle_reports():
+                # For overall bundle size
+                if MeasurementName.bundle_analysis_report_size.value in dataset_names:
+                    self._save_to_timeseries(
+                        db_session,
+                        commit,
+                        MeasurementName.bundle_analysis_report_size.value,
+                        bundle_report.name,
+                        bundle_report.total_size(),
+                    )
+
+                # For individual javascript associated assets using UUID
+                if MeasurementName.bundle_analysis_asset_size.value in dataset_names:
+                    for asset in bundle_report.asset_reports():
+                        if asset.asset_type == AssetType.JAVASCRIPT:
```

Comment:
is this going to be `== to AssetType.JAVASCRIPT` or `== to AssetType.JAVASCRIPT.value`? Same for the other enums

Reply:
no `.value` here because the `asset_type` is a `SQLAlchemyEnum` type which does the comparison without getting the actual values